modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/OpenAPIResourceTest.java

```diff
@@ -104,6 +118,100 @@ public void testGetOpenAPI() throws Exception {
 		Assert.assertEquals("NOT_FOUND", jsonObject.getString("status"));
 	}
 
+	private String _getNestedEntitySchema(
+		JSONObject jsonObject, ObjectRelationship objectRelationship,
+		ObjectDefinition objectDefinition) {
+
+		String nestedEntitySchema;
+
+		JSONObject nestedEntitySchemaJSONObject = jsonObject.getJSONObject(
+			"components"
+		).getJSONObject(
+			"schemas"
+		).getJSONObject(
+			objectDefinition.getShortName()
+		).getJSONObject(
+			"properties"
+		).getJSONObject(
+			objectRelationship.getName()
+		);
+
+		if (Objects.equals(
+				objectRelationship.getType(),
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY) &&
+			(objectDefinition.getObjectDefinitionId() ==
+				_objectDefinition2.getObjectDefinitionId())) {
+
+			nestedEntitySchema = (String)nestedEntitySchemaJSONObject.get(
+				"$ref");
+		}
+		else {
+			nestedEntitySchema =
+				(String)nestedEntitySchemaJSONObject.getJSONObject(
+					"items"
+				).get(
+					"$ref"
+				);
+		}
+
+		String[] nestedEntitySchemaSplit = nestedEntitySchema.split("/");
+
+		return nestedEntitySchemaSplit[nestedEntitySchemaSplit.length - 1];
+	}
+
+	private void _testGetNestedEntityInObjectRelationship(
+			String objectRelationshipType)
+		throws Exception {
+
+		_objectDefinition2 = ObjectDefinitionTestUtil.publishObjectDefinition(
+			Collections.singletonList(
+				ObjectFieldUtil.createObjectField(
+					"Text", "String", true, true, null,
+					RandomTestUtil.randomString(), _OBJECT_FIELD_NAME, false)));
+
+		ObjectRelationship objectRelationship =
+			ObjectRelationshipTestUtil.addObjectRelationship(
+				_objectDefinition1, _objectDefinition2,
+				TestPropsValues.getUserId(), objectRelationshipType);
+
+		JSONObject jsonObject = HTTPTestUtil.invoke(
```

Comment:
I am not sure if we should test this here as we are testing that the nested entities are included as a field in the schema

Reply:
I implemented the tests here, because if we need to assure that nested entities are included in the schema, we must operate with its OpenAPI